src/main/java/duke/storage/Storage.java

```diff
@@ -0,0 +1,132 @@
+package duke.storage;
+
+import duke.task.*;
+import duke.utilities.DukeException;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Storage class for loading nd saving data from a file
+ */
+public class Storage {
+    private final File file;
+
+    /**
+     * Construct a new Storage object with a given file path.
+     * Ensures the file is accessible.
+     *
+     * @param filePath The path to the file to store data in.
+     * @throws IOException Handles case of File not being found or location not existing.
+     */
+    public Storage(String filePath) throws IOException {
+        String[] parts = filePath.split("/");
+
+        File folder = new File(parts[0]);
+        if (!folder.exists()) {
+            folder.mkdir();
+        }
+
+        File file = new File(filePath);
+        if (!file.exists()) {
+            file.createNewFile();
+        }
+
+        this.file = file;
+    }
+
+    /**
+     * Saves the {@code Task}s from a {@code TaskList} to the {@code file}.
+     * @param taskList The {@code TaskList} to save.
+     * @throws IOException Handles File not being found when writing to hard disk.
+     */
+    public void writeTasksToStorage(TaskList taskList) throws IOException {
+        ArrayList<Task> tasks = taskList.getTasks();
+        FileWriter fileWriter = new FileWriter(this.file);
+
+        for (Task task : tasks) {
+            String taskCode = task.toString().substring(0,3);
+            switch (taskCode) {
+            case "[T]": {
+                String output = "T|" + task.getStatusIcon() + "|" + task.getDescription();
+                fileWriter.write(output);
+                break;
+            }
```

Comment:
I like that there is no indentation for case clauses. Just a few nits, I don't think that the curly brackets are necessary after case clauses